portal-impl/src/com/liferay/portal/zip/LarZipWriterImpl.java

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.zip;
+
+import com.liferay.petra.io.StreamUtil;
+import com.liferay.portal.util.PropsValues;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+
+import java.util.AbstractMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Minhchau Dang
+ */
+public class LarZipWriterImpl extends ZipWriterImpl {
```

Comment:
@holatuwol @hongvo2308 This should not be a type of ZipWritter, it should be a wrapper of ZipWritter which will auto flush when exceeding ZIP_FILE_WRITER_EXPORT_BUFFER_SIZE.